coriolis/tests/conductor/rpc/data/deploy_replica_instance_config.yml

```diff
@@ -0,0 +1,118 @@
+      depends_on: ['VALIDATE_REPLICA_DEPLOYMENT_INPUTS']
+    - type: 'DEPLOY_REPLICA_INSTANCE_RESOURCES'
+      depends_on: ['CREATE_REPLICA_DISK_SNAPSHOTS']
+    - type: 'DEPLOY_OS_MORPHING_RESOURCES'
+      depends_on: ['DEPLOY_REPLICA_INSTANCE_RESOURCES']
+    - type: 'OS_MORPHING'
+      depends_on: ['DEPLOY_OS_MORPHING_RESOURCES']
+    - type: 'DELETE_OS_MORPHING_RESOURCES'
+      depends_on: ['DEPLOY_OS_MORPHING_RESOURCES', 'OS_MORPHING']
+      on_error: True
+    - type: 'FINALIZE_REPLICA_INSTANCE_DEPLOYMENT'
+      depends_on: ['OS_MORPHING', 'DELETE_OS_MORPHING_RESOURCES']
+    - type: 'DELETE_REPLICA_TARGET_DISK_SNAPSHOTS'
+      depends_on: ['CREATE_REPLICA_DISK_SNAPSHOTS', 'FINALIZE_REPLICA_INSTANCE_DEPLOYMENT']
+      on_error: False
+    - type: 'CLEANUP_FAILED_REPLICA_INSTANCE_DEPLOYMENT'
+      depends_on: ['DEPLOY_REPLICA_INSTANCE_RESOURCES', 'FINALIZE_REPLICA_INSTANCE_DEPLOYMENT']
+      on_error_only: True
+    - type: 'RESTORE_REPLICA_DISK_SNAPSHOTS'
+      depends_on: ['CLEANUP_FAILED_REPLICA_INSTANCE_DEPLOYMENT']
+      on_error: True
+
+- config:
+    skip_os_morphing: True
+  expected_tasks:
+    - type: 'CREATE_REPLICA_DISK_SNAPSHOTS'
+      depends_on: ['VALIDATE_REPLICA_DEPLOYMENT_INPUTS']
+    - type: 'DEPLOY_REPLICA_INSTANCE_RESOURCES'
+      depends_on: ['CREATE_REPLICA_DISK_SNAPSHOTS']
+    - type: 'FINALIZE_REPLICA_INSTANCE_DEPLOYMENT'
+      depends_on: ['DEPLOY_REPLICA_INSTANCE_RESOURCES']
+    - type: 'DELETE_REPLICA_TARGET_DISK_SNAPSHOTS'
+      depends_on: ['CREATE_REPLICA_DISK_SNAPSHOTS', 'FINALIZE_REPLICA_INSTANCE_DEPLOYMENT']
+      on_error: False
+    - type: 'CLEANUP_FAILED_REPLICA_INSTANCE_DEPLOYMENT'
+      depends_on: ['DEPLOY_REPLICA_INSTANCE_RESOURCES', 'FINALIZE_REPLICA_INSTANCE_DEPLOYMENT']
+      on_error_only: True
+    - type: 'RESTORE_REPLICA_DISK_SNAPSHOTS'
+      depends_on: ['CLEANUP_FAILED_REPLICA_INSTANCE_DEPLOYMENT']
+      on_error: True
+
+- config:
+    skip_os_morphing: True
+    get_optimal_flavor: True
+  expected_tasks:
+    - type: 'CREATE_REPLICA_DISK_SNAPSHOTS'
+      depends_on: ['VALIDATE_REPLICA_DEPLOYMENT_INPUTS']
+    - type: 'DEPLOY_REPLICA_INSTANCE_RESOURCES'
+      depends_on: ['CREATE_REPLICA_DISK_SNAPSHOTS']
+    - type: 'GET_OPTIMAL_FLAVOR'
+      depends_on: ['DEPLOY_REPLICA_INSTANCE_RESOURCES']
+    - type: 'FINALIZE_REPLICA_INSTANCE_DEPLOYMENT'
+      depends_on: ['GET_OPTIMAL_FLAVOR']
+    - type: 'DELETE_REPLICA_TARGET_DISK_SNAPSHOTS'
+      depends_on: ['CREATE_REPLICA_DISK_SNAPSHOTS', 'FINALIZE_REPLICA_INSTANCE_DEPLOYMENT']
+      on_error: False
+    - type: 'CLEANUP_FAILED_REPLICA_INSTANCE_DEPLOYMENT'
+      depends_on: ['DEPLOY_REPLICA_INSTANCE_RESOURCES', 'FINALIZE_REPLICA_INSTANCE_DEPLOYMENT']
+      on_error_only: True
+    - type: 'RESTORE_REPLICA_DISK_SNAPSHOTS'
+      depends_on: ['CLEANUP_FAILED_REPLICA_INSTANCE_DEPLOYMENT']
+      on_error: True
+
+- config:
+    skip_os_morphing: True
+    clone_disks: True
```

Comment:
Granted that only one case of `clone_disks: True` is sufficient for this test, only one case of `get_optimal_flavor: True` is not. Since this flag adds a task in the middle of the execution, and also changes other tasks' `depends_on`, please also add expected tasks for these configs:

`get_optimal_flavor: True, skip_os_morphing: False` & `get_optimal_flavor: True, skip_os_morphing: False, has_os_morphing_minion: True`